pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -697,16 +701,7 @@ public void start() throws PulsarServerException {
             this.brokerAdditionalServlets = AdditionalServlets.load(config);
 
             this.webService = new WebService(this);
-            this.metricsServlet = new PrometheusMetricsServlet(
-                    this, config.isExposeTopicLevelMetricsInPrometheus(),
-                    config.isExposeConsumerLevelMetricsInPrometheus(),
-                    config.isExposeProducerLevelMetricsInPrometheus(),
-                    config.isSplitTopicAndPartitionLabelInPrometheus());
-            if (pendingMetricsProviders != null) {
-                pendingMetricsProviders.forEach(provider -> metricsServlet.addRawMetricsProvider(provider));
-                this.pendingMetricsProviders = null;
-            }
-
+            createMetricsServlet();
```

Comment:
below we have a non synchronised access to metricsServlet.
I suggest to use a `synchronised` getter and use it while calling `addWebServerHandlers`